### PR TITLE
Vet Center main page - Locations and contact information section (address, image, static map, hours)

### PR DESCRIPTION
**Pull request by brangi**
## Description
issue 
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22391
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22393

### CODE REVIEW

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      </div>
+    </div>
+  );
+};
+
+VetCenterHours.propTypes = {
+  hoursMap: PropTypes.array,
+};
+
+export default VetCenterHours;
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:69]: Hmmm - I thought we decided to use `src/site/paragraphs/service_hours.drupal.liquid`?

Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:69]: That template uses a `<table>` to align the columns. We'll want to do something similar here, I think.

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js [hunk 2]
```diff
@@ -63,10 +86,9 @@ const vetCenterFragment = `
 const GetVetCenters = `
   ${vetCenterFragment}
   
-  query GetVetCenters($onlyPublishedContent: Boolean!) {
```
Comment (mpelzsherman) [inline src/site/stages/build/drupal/graphql/vetCenter.graphql.js:66]: We should not check in this change!

Maybe we can come up with a way to toggle this using a command-line switch in the build?

File: src/applications/static-pages/facilities/createFacilityPage.js [hunk 3]
```diff
@@ -22,7 +22,8 @@ export default async function createFacilityPage(store) {
     facilityID = JSON.parse(detailWidget.dataset.facility);
   }
   if (mapWidget && !facilityID) {
-    facilityID = JSON.parse(mapWidget.dataset.facility);
+    facilityID =
+      mapWidget.dataset.facility || JSON.parse(mapWidget.dataset.facility);
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/createFacilityPage.js:26]: Why is this change needed?

Comment (mpelzsherman) [inline src/applications/static-pages/facilities/createFacilityPage.js:26]: Can you add a comment explaning why this is needed?

File: src/site/includes/vet_center_hours.liquid [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<script nonce="**CSP_NONCE**" type="text/javascript">
+    window.vetCenterHours = {{ vetCenterHours | json }};
+</script>
+<div data-widget-type="vet-center-hours"></div>
```
Comment (mpelzsherman) [inline src/site/includes/vet_center_hours.liquid:4]: Instead of passing the vetCenterHours data via the `window` global, how about passing it as a data attribuge on the div, e.g.

```suggestion
<div data-widget-type="vet-center-hours"
        data-hours="{{ vetCenterHours | json }}"
></div>
```

We might not even need this file if we take that approach.

File: src/applications/static-pages/facilities/vetCentersHours.jsx (hunk 1 above)
Comment (brangi) [reply src/applications/static-pages/facilities/vetCentersHours.jsx:69]: I mentioned this yesterday in standup, I decided to use react because it's not the same data strcuture, it's easier and more flexible to use react

**Comment by mpelzsherman** [comment]
The subheadings under "Locations and contact information" are larger than the prototype, but they are consistent with VAMC's. Please check with @lildziner.

File: src/site/stages/build/drupal/graphql/vetCenter.graphql.js (hunk 2 above)
Comment (brangi) [reply src/site/stages/build/drupal/graphql/vetCenter.graphql.js:66]: ok

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      </h3>
+      <div className="vads-u-display--flex vads-u-flex-direction--column small-screen:vads-u-flex-direction--row vads-u-margin-bottom--0">
+        {buildHours(props.hoursMap)}
+      </div>
+    </div>
+  );
+};
+
+VetCenterHours.propTypes = {
+  hoursMap: PropTypes.array,
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:66]: Why call this a "map" if it's an array?

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    let AMorPM = 'a.m.';
+    if (hour >= 0 && hour <= 24 && minutes >= 0 && minutes <= 60) {
+      if (hour > 12) AMorPM = 'p.m.';
+      hour %= 12;
+    }
+    return `${hour}:${
+      minutes === 0 ? `${minutes}${minutes}` : `${minutes}`
+    } ${AMorPM}`;
+  };
+  const buildHours = hours => {
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:29]: Can we please add a newline above each function declaration?

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// TODO fix spacing and add unit tests
+const VetCenterHours = props => {
+  const arrayOfWeekdays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+
+  const formatTime = hoursString => {
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:8]: We should definitely write a unit test for this function. It probably should live in `src/applications/facility-locator/utils`.

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        ? hoursString.toString().substring(0, 2)
+        : hoursString.toString().substring(0, 1),
+    );
+    const minutes = Number(
+      hoursString.toString().length > 3
+        ? hoursString.toString().substring(2, 4)
+        : hoursString.toString().substring(1, 4),
+    );
+    let AMorPM = 'a.m.';
+    if (hour >= 0 && hour <= 24 && minutes >= 0 && minutes <= 60) {
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:21]: This line should always return true, otherwise the data is malformed, and we should probably just render nothing, or maybe throw a build error? Need to ask the CMS team perhaps.

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+      if (hour > 12) AMorPM = 'p.m.';
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:22]: If we return `null` after failing the sanity check on line 21, I think we can use a const instead of a let here:

```suggestion
      const AMorPM = hour > 12 ? 'a.m.' : 'p.m.';
```

File: src/applications/static-pages/facilities/vetCentersHours.jsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      <ul className="vads-u-flex--1 va-c-facility-hours-list vads-u-margin-top--0 vads-u-margin-bottom--1 small-screen:vads-u-margin-bottom--0 vads-u-margin-right--3">
+        {hoursListItems}
+      </ul>
+    );
+  };
+
+  return (
+    <div className="vads-u-margin-bottom--3" id="vet-center-hours">
+      <h3 className="vads-u-font-size--lg vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">
+        Hours
```
Comment (mpelzsherman) [inline src/applications/static-pages/facilities/vetCentersHours.jsx:56]: If `hoursMap` is empty, should we render anything?

File: src/site/layouts/vet_center.drupal.liquid [hunk 11]
```diff
@@ -26,6 +26,66 @@
...
+                                </div>
+                                <h3 class="vads-u-font-size--lg vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">Direct line</h3>
+                                <div class="vads-u-margin-bottom--3">
+                                    <div class="main-phone vads-u-margin-bottom--1">
+                                        </strong><a href="tel:{{ fieldPhoneNumber }}">{{ fieldPhoneNumber }}</a>
+                                    </div>
+                                </div>
+                               <div class="vads-u-margin-bottom--3">
+                                   <h3 class="vads-u-font-size--lg vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">Other locations</h3>
+                                   <p>Vet Centers are community based to be more <br>
```
Comment (mpelzsherman) [inline src/site/layouts/vet_center.drupal.liquid:56]: Rather than adding <br> tags here, can we use CSS instead to wrap the text as desired?

File: src/applications/facility-locator/tests/helpers/formatHours.unit.spec.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  it('Should return a null value 0', () => {
+    expect(formatHours(0)).to.equal(null);
+  });
+
+  it('Should return a null value 44', () => {
+    expect(formatHours(44)).to.equal(null);
+  });
+});
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/helpers/formatHours.unit.spec.js:28]: Nice job on the tests!

File: src/applications/facility-locator/utils/formatHours.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  ) {
+    const AMorPM = hour > 12 ? 'p.m.' : 'a.m.';
+    hour %= 12;
+    return `${hour}:${
+      minutes === 0 ? `${minutes}${minutes}` : `${minutes}`
+    } ${AMorPM}`;
+  }
+
+  return null;
+};
```
Comment (mpelzsherman) [inline src/applications/facility-locator/utils/formatHours.js:30]: 🚀

File: src/site/layouts/vet_center.drupal.liquid [hunk 14]
```diff
@@ -26,6 +26,66 @@
...
+                                 alt="{{ image.alt }}" title="{{ image.title }}"
+                                 width="{{ image.derivative.width }}"
+                                 height="{{ image.derivative.height }}" />
+                        </a>
+                    {% endif %}
+                  <!-- facility map -->
+                  <div data-widget-type="facility-map"
+                       data-facility="{{ fieldFacilityLocatorApiId }}">
+                  </div>
+              </div>
```
Comment (mpelzsherman) [inline src/site/layouts/vet_center.drupal.liquid:86]: Lines 69:86 are an exact duplicate of [this](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/layouts/health_care_local_facility.drupal.liquid#L114-L130). Can we refactor?

File: src/applications/facility-locator/tests/helpers/formatHours.unit.spec.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+    expect(formatHours(700)).to.equal('7:00 a.m.');
+  });
+
+  it('Should return a formatted hour string 5:00 p.m.', () => {
+    expect(formatHours(1700)).to.equal('5:00 p.m.');
+  });
+
+  it('Should return a null value for 9000', () => {
+    expect(formatHours(9000)).to.equal(null);
+  });
```
Comment (mpelzsherman) [inline src/applications/facility-locator/tests/helpers/formatHours.unit.spec.js:19]: Maybe add a few more malformed tests, e.g. 2401 or 0061.

File: src/site/layouts/health_care_local_facility.drupal.liquid [hunk 16]
```diff
@@ -111,23 +111,11 @@
...
-                  <img class="facility-img" src="{{ image.derivative.url }}"
-                       alt="{{ image.alt }}" title="{{ image.title }}"
-                       width="{{ image.derivative.width }}"
-                       height="{{ image.derivative.height }}" />
-                </a>
-              {% endif %}
-              <div data-widget-type="facility-map"
-                   data-facility="{{ fieldFacilityLocatorApiId | widgetFacilityDetail | escape }}">
-              </div>
-            </div>
```
Comment (mpelzsherman) [inline src/site/layouts/health_care_local_facility.drupal.liquid:130]: ❤️

File: src/site/layouts/vet_center.drupal.liquid [hunk 17]
```diff
@@ -26,6 +26,53 @@
...
+                                         target="_blank" rel="noopener noreferrer">Directions
+                                        </a>
+                                    </div>
+                                </div>
+                                <h3 class="vads-u-font-size--lg vads-u-margin-top--0 vads-u-line-height--1 vads-u-margin-bottom--1">Direct line</h3>
+                                <div class="vads-u-margin-bottom--3">
+                                    <div class="main-phone vads-u-margin-bottom--1">
+                                        </strong><a href="tel:{{ fieldPhoneNumber }}">{{ fieldPhoneNumber }}</a>
+                                    </div>
+                                </div>
```
Comment (mpelzsherman) [inline src/site/layouts/vet_center.drupal.liquid:53]: Here's another refactor/reuse opportunity. This `<section class="vads-facility-detail">` block is copied/pasted form `src/site/layouts/health_care_local_facility.drupal.liquid`. Maybe we can clean this up later?

**Review by mpelzsherman — APPROVED** [review]

**Review by jhonny218 — APPROVED** [review]
LGTM
